### PR TITLE
Step patterns

### PR DESCRIPTION
**Pull request by garthwebb**
Two new step patterns, plus some small fixes.

Fixes - Reduced devserver max LED value to 255 since I don't think the web interface can do 10-bit color.   Was not seeing any intensity changes above values of 255 in any case.

Patterns:

### SimpleStep

Lights the square that's being stood on.  Hue, value and saturation linked to APCmini controls

### HeatStep

Standing on a square "heats" it up slowly.  A competing "cooling" factor cools the floor back down when the foot is released, and keeps a held step from heating up the whole floor.

APCmini controls for:
* step hue
* heat hue
* overall saturation
* heat factor
* cool factor

### CODE REVIEW

File: floor/floor/driver/devserver.py [hunk 1]
```diff
@@ -70,6 +70,9 @@ def serve_forever(port=1979):
 
 class Devserver(Base):
     """Floor driver interface."""
+
+    MAX_LED_VALUE = 255
```
Comment (garthwebb) [inline floor/floor/driver/devserver.py:74]: On my browser, seems like 8-bit color channels is the max

File: floor/floor/processor/heat_step.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        self.secondary_hue = 0.5
+        self.saturation = 1.0
+        self.value = 1.0
+
+        self.heat_factor = 20.5
+        self.cool_factor = 0.95
+        self.heat_delta = self.HEAT_RANGE[1] - self.HEAT_RANGE[0]
+        self.cool_delta = self.COOL_RANGE[1] - self.COOL_RANGE[0]
+
+    def handle_midi_command(self, command):
```
Comment (mik3y) [inline floor/floor/processor/heat_step.py:44]: This gives me an idea - we could maybe define a handful of abstract "control knobs" (like "aux1", "aux2", "aux3", "aux4") and wire them up to midi in the controller (but also the web ui). Let the processor plumb them through in the RenderContext. Each processor can then decide what, if anything, it does with eg "the current value of aux3", just like they do with things like weights. Basically - make it a bit simpler & midi-free down here in the processors.

**Review by mik3y — APPROVED** [review]
oooo these new patterns look awesome!

File: floor/floor/processor/heat_step.py (hunk 2 above)
Comment (garthwebb) [reply floor/floor/processor/heat_step.py:44]: haha, yah, I just started another change set that does this.  Let me get the v1 into a PR and see what you think.

**Pull request merged**